app/assets/scripts/components/upload-form.js

```diff
@@ -62,9 +62,8 @@ var UploadForm = React.createClass({
           failures = ['No data provided'];
           this.logOutput(failures);
         }
-        // Check header on line 1
+        // Check header on first line
         if (line === 0) this.checkHeader(data, failures);
```

Comment:
You're passing the `failures` array around to be freely modified by other functions. You should try to avoid this as it's very difficult to keep track of the result of the function. The function should instead return values and they should be added to the main array.